src/g2.rs

```diff
@@ -595,6 +598,19 @@ impl VariableBaseMSM for G2Projective {
     }
 }
 
+#[cfg(any(feature = "cuda", feature = "opencl"))]
+impl VariableBaseMSM for G2Projective {
+    fn msm(bases: &[Self::MulBase], exponents: &[Self::ScalarField]) -> Result<Self, usize> {
+        let bigints = exponents
+            .iter()
+            .map(|exponent| exponent.into_bigint())
```

Comment:
Not a _big_ deal but into_bigint() eventually pass from num_biguint and that does some transformation - it's not just "reading the bytes", so here we pay a linear cost before having just the raw bytes. We could just call `to_bytes_le()` and map to what your msm gpu code expects (I guess stg like `[u64]`? if so we can even take the `blst_fr.l` field ?)

Reply:
Nope, it needs to be that transform, not just the raw bytes.

Reply:
In case of `blstrs` we need to call `to_repr()` on the exponents.